diff -rupN ../binutils-tmp/binutils-2.24/bfd/config.bfd binutils-2.24/bfd/config.bfd
--- ../binutils-tmp/binutils-2.24/bfd/config.bfd	2013-11-04 10:33:37.000000000 -0500
+++ binutils-2.24/bfd/config.bfd	2015-02-25 14:30:10.000000000 -0500
@@ -715,6 +715,10 @@ case "${targ}" in
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs="i386pe_vec i386pei_vec"
     ;;
+  i[3-7]86-*-hydrax*)
+    targ_defvec=bfd_elf32_i386_vec
+    targ_selvecs=
+    ;;
   i[3-7]86-*-interix*)
     targ_defvec=i386pei_vec
     targ_selvecs="i386pe_vec"
diff -rupN ../binutils-tmp/binutils-2.24/config.sub binutils-2.24/config.sub
--- ../binutils-tmp/binutils-2.24/config.sub	2013-11-26 06:37:33.000000000 -0500
+++ binutils-2.24/config.sub	2015-02-25 14:30:10.000000000 -0500
@@ -1376,7 +1376,7 @@ case $os in
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -hydrax*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -rupN ../binutils-tmp/binutils-2.24/gas/configure.tgt binutils-2.24/gas/configure.tgt
--- ../binutils-tmp/binutils-2.24/gas/configure.tgt	2013-11-04 10:33:37.000000000 -0500
+++ binutils-2.24/gas/configure.tgt	2015-02-25 14:30:10.000000000 -0500
@@ -206,6 +206,7 @@ case ${generic_target} in
   i386-*-beos*)				fmt=elf ;;
   i386-*-coff)				fmt=coff ;;
   i386-*-elf)				fmt=elf ;;
+  i386-*-hydrax*)			fmt=elf ;;
   i386-*-kaos*)				fmt=elf ;;
   i386-*-bsd*)				fmt=aout em=386bsd ;;
   i386-*-nacl*)				fmt=elf em=nacl
diff -rupN ../binutils-tmp/binutils-2.24/ld/configure.tgt binutils-2.24/ld/configure.tgt
--- ../binutils-tmp/binutils-2.24/ld/configure.tgt	2013-11-26 06:37:33.000000000 -0500
+++ binutils-2.24/ld/configure.tgt	2015-02-25 14:30:10.000000000 -0500
@@ -223,6 +223,8 @@ x86_64-*-rdos*)		targ_emul=elf64rdos ;;
 i[3-7]86-*-bsd)		targ_emul=i386bsd ;;
 i[3-7]86-*-bsd386)	targ_emul=i386bsd ;;
 i[3-7]86-*-bsdi*)	targ_emul=i386bsd ;;
+i[3-7]86-*-hydrax*)	targ_emul=elf_i386_hydrax
+			targ_extra_emuls=elf_i386 ;;
 i[3-7]86-*-aout)	targ_emul=i386aout ;;
 i[3-7]86-*-linux*aout*)	targ_emul=i386linux
 			targ_extra_emuls=elf_i386
diff -rupN ../binutils-tmp/binutils-2.24/ld/emulparams/elf_i386_hydrax.sh binutils-2.24/ld/emulparams/elf_i386_hydrax.sh
--- ../binutils-tmp/binutils-2.24/ld/emulparams/elf_i386_hydrax.sh	1969-12-31 19:00:00.000000000 -0500
+++ binutils-2.24/ld/emulparams/elf_i386_hydrax.sh	2015-02-25 14:30:10.000000000 -0500
@@ -0,0 +1,3 @@
+. ${srcdir}/emulparams/elf_i386.sh
+GENERATE_SHLIB_SCRIPT=yes
+GENERATE_PIE_SCRIPT=yes
diff -rupN ../binutils-tmp/binutils-2.24/ld/Makefile.am binutils-2.24/ld/Makefile.am
--- ../binutils-tmp/binutils-2.24/ld/Makefile.am	2013-11-26 06:37:33.000000000 -0500
+++ binutils-2.24/ld/Makefile.am	2015-02-25 14:30:10.000000000 -0500
@@ -263,6 +263,7 @@ ALL_EMULATION_SOURCES = \
 	eelf32xstormy16.c \
 	eelf32xtensa.c \
 	eelf_i386.c \
+	eelf_i386_hydrax.c \
 	eelf_i386_be.c \
 	eelf_i386_chaos.c \
 	eelf_i386_fbsd.c \
@@ -1268,6 +1269,11 @@ eelf32xtensa.c: $(srcdir)/emulparams/elf
 eelf_i386.c: $(srcdir)/emulparams/elf_i386.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386 "$(tdir_elf_i386)"
+
+eelf_i386_hydrax.c: $(srcdir)/emulparams/elf_i386_hydrax.sh \
+  $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+	${GENSCRIPTS} elf_i386_hydrax "$(tdir_elf_i386_hydrax)"
+
 eelf_i386_be.c: $(srcdir)/emulparams/elf_i386_be.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386_be "$(tdir_elf_i386_be)"
